docs/user-guide/src/bmo/bmo.md

```diff
@@ -0,0 +1,44 @@
+<h1>Bare Metal Operator</h1>
+
+The Bare Metal Operator (BMO) is a custom Kubernetes controller that reconciles BareMetalHost resources using information provided by Ironic and the Cluster API Provider Metal3 (CAPM3) controller. A simplified information flow path is shown below:
+
+```text
+CAPM3 ----------> BMO <----------> Ironic <----------> Server
+            BareMetalHost resource
+            - server MACAddress                        - image
+            - server address                           - cloud-init data
+            - server credentials
+```
+
+It would help to use an example to describe what BMO does. There are two operations of interest, getting hardware details of the server and  booting the server with a given image, including user supplied cloud-init data. The BareMetalHost resource contains address and authentication information towards a server.
+
+BMO communicates this information to Ironic and gets hardware details, such as CPU and disk, of the server in return. This information is added to the BareMetalHost resource. In order to get such server related information, the server is booted with IPA image.
+If there are hardware related changes, the BareMetalHost is updated accordingly.
+
+```text
+CAPM3 ----------> BMO <----------> Ironic <----------> Server
+            BareMetalHost resource
+            - server MACAddress                        - IPA-image
+            - server address
+            - server credentials
+            - Hardware details [cpu,disk,...]
+```
+
+Next, with the information combing from the CAPM3 side, BMO updates the BareMetalHost with image and cloud-init data. That information is also conveyed to Ironic and the server is Booted accordingly.
```

Comment:
BMO does not update these, you probably meant "CAPM3 updates"? Or you meant that BMO requests something from ironic?

nit: no need to upper-case "Booted"

Reply:
I was trying to simplify the process. changed it to a passive form.